drivers/sensor/stm32_temp/stm32_temp.c

```diff
@@ -86,10 +111,17 @@ static const struct sensor_driver_api stm32_temp_driver_api = {
 static int stm32_temp_init(const struct device *dev)
 {
 	struct stm32_temp_data *data = dev->data;
+	const struct stm32_temp_config *cfg = dev->config;
 	struct adc_channel_cfg *accp = &data->adc_cfg;
 	struct adc_sequence *asp = &data->adc_seq;
 	int rc;
 
+	if (cfg->has_cal && (!cfg->cal_vrefanalog || !cfg->cal1_addr || !cfg->cal2_addr ||
+			     cfg->cal1_temp >= cfg->cal2_temp)) {
+		LOG_ERR("Invalid configuration");
+		return -EINVAL;
+	}
```

Comment:
I think that dt tools should prevent misconfiguration, so might be useless.

Reply:
you still can assign 0 to all ot these, which would case division by 0 or null pointer referenc. The checks are not covered by dt tools, or at leat I couldn't find a range option in the dts bindings, so I think it's safer to keep them if not possible in the dts binding?